src/views/SceneView.js

```diff
@@ -19,6 +19,7 @@ export default class SceneView extends PureComponent<void, Props, void> {
 
   render() {
     const { screenProps, navigation, component: Component } = this.props;
-    return <Component {...screenProps} navigation={navigation} />;
+    // pass through screenProps as well for nested navigators
+    return <Component {...screenProps} screenProps={screenProps} navigation={navigation} />;
```

Comment:
One other option would be to only pass the non-splatted version if Component is a navigator, though that may require type introspection or forking the behavior further up the stack.

My gut says that making the breaking change is probably the best option since it is the cleanest, but I also agree that avoiding breaking changes is probably for the better.